### PR TITLE
Add script to generate new package

### PR DESCRIPTION
**Pull request by ghost**
I'll have to make another PR for the `master` branch making the same change to the README.

### CODE REVIEW

**Comment by flying-sheep** [comment]
Wow, you sure went quickly from “sounds like a good idea” to “here’s a working implementation”, thank you!

FWIW: I tried the script and it worked well.

File: README.md [hunk 1]
```diff
@@ -109,43 +109,15 @@ Your package should have this structure:
...
-    },
-    "files": [
-        "index.d.ts",
-        "foo-tests.ts"
-    ]
-}
+```sh
+npm install
+npm run compile-scripts
+node scripts/new-package.js new-package-name
```
Comment (mhegazy) [inline README.md]: consider checking in the .js file and making the script an npm script. e.g. `npm run new-package new-package-name`

**Review by flying-sheep — COMMENTED** [review]
you ignored my review here. don’t you think my additions would be valuable?

**Comment by andy-ms** [comment]
We have `npm run compile-scripts` and also now `npm run new-package`, what more should there be?

File: scripts/new-package.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import * as path from "path";
+
+const newPackageName = process.argv[2];
+if (!newPackageName) {
+    throw new Error("Usage: node scripts/new-package.js new-package-name")
+}
+
+mkdirSync(newPackageName);
+
+write("index.d.ts", `// Type definitions for ${newPackageName} 1.2
```
Comment (andy-ms) [inline scripts/new-package.ts]: The version should match what's on NPM. (For non-NPM packages, hopefully they have some usable semver.) Could you make a PR for that script?

**Comment by flying-sheep** [comment]
oh, nothing. i just meant the version thing above.

sorry for the confusion!

File: scripts/new-package.ts (hunk 2 above)
Comment (flying-sheep) [inline scripts/new-package.ts]: OK!

**Pull request merged**